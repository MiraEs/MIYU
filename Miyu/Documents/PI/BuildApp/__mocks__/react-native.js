/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
