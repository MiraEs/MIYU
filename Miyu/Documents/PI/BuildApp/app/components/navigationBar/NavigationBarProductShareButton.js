import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import branch from 'react-native-branch';
import Share from 'react-native-share';
import NavigationBarIconButton from './NavigationBarIconButton';
import TrackingActions from '../../lib/analytics/TrackingActions';
import helpers from '../../lib/helpers';

export class NavigationBarProductShareButton extends Component {

	onSharePress = async () => {
		const {
			manufacturer,
			title,
			productId,
			compositeId,
			type,
			selectedFinish,
			uniqueId,
			useBranchLinksForShare,
		} = this.props;
		let url = `https://www.build.com/${helpers.slugify(`${manufacturer}-${productId}`)}/s${compositeId}${uniqueId ? `?uid=${uniqueId}` : ''}`;
		if (useBranchLinksForShare) {
			const branchUniversalObject = await branch.createBranchUniversalObject(`${manufacturer} ${productId} ${title}`, {
				title: `${manufacturer} ${productId} ${title} on Build.com`,
				contentIndexingMode: 'private',
			});
			const linkProperties = {
				feature: 'product page share',
				channel: 'BuildApp',
			};
			const controlParams = {
				$canonical_url: url,
				$desktop_url: url,
				$android_url: url,
				$ios_url: url,
			};
			const shortLink = await branchUniversalObject.generateShortUrl(linkProperties, controlParams);
			url = shortLink.url;
		}
		Share.open({
			message: `View the ${manufacturer} ${productId} ${title} on Build.com`,
			title: 'Share Product',
			url,
		}).then((result) => {
			tracking.trackProductShared(result, {
				manufacturer,
				title,
				productId,
				compositeId,
				type,
				selectedFinish,
			});
		}).catch(helpers.noop).done();

	};

	render() {
		const {
			description,
		} = this.props;
		return (
			<NavigationBarIconButton
				onPress={this.onSharePress}
				iconName={helpers.getIcon('share')}
				loading={!description}
				trackAction={TrackingActions.PDP_NAV_TAP_SHARE}
			/>
		);
	}

}

NavigationBarProductShareButton.propTypes = {
	manufacturer: PropTypes.string,
	title: PropTypes.string,
	productId: PropTypes.string,
	compositeId: PropTypes.number,
	type: PropTypes.string,
	selectedFinish: PropTypes.object,
	uniqueId: PropTypes.number,
	useBranchLinksForShare: PropTypes.bool,
	description: PropTypes.string,
};

NavigationBarProductShareButton.defaultProps = {

};

export const mapStateToProps = (state, ownProps) => {
	const product = state.productsReducer[ownProps.compositeId];
	if (!product) {
		return {
			description: '',
		};
	}
	const {
		manufacturer,
		title,
		productId,
		type,
		selectedFinish,
		description,
	} = product;
	let uniqueId;
	if (selectedFinish && selectedFinish.uniqueId) {
		uniqueId = selectedFinish.uniqueId;
	}
	return {
		useBranchLinksForShare: state.featuresReducer.features.useBranchLinksForShare,
		manufacturer,
		title,
		productId,
		type,
		selectedFinish,
		uniqueId,
		description,
	};
};

export default connect(mapStateToProps)(NavigationBarProductShareButton);
