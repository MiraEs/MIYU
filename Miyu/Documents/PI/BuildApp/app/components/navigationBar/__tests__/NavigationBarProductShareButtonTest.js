jest.unmock('react-native');
jest.mock('prop-types');
jest.mock('react-redux');
jest.mock('react-native-branch', () => ({
	createBranchUniversalObject: jest.fn(() => ({
		generateShortUrl: jest.fn(() => ({
			url: '/url',
		})),
	})),
}));
import branch from 'react-native-branch';
jest.mock('react-native-share', () => ({
	open: jest.fn(() => ({
		then: jest.fn(() => ({
			catch: jest.fn(() => ({
				done: jest.fn(),
			})),
		})),
	})),
}));
import Share from 'react-native-share';
jest.mock('../NavigationBarIconButton', () => 'NavigationBarIconButton');
jest.mock('../../../lib/analytics/TrackingActions', () => ({
	PDP_NAV_TAP_SHARE: 'PDP_NAV_TAP_SHARE',
}));
jest.mock('../../../lib/helpers', () => ({
	getIcon: jest.fn((name) => name),
	slugify: jest.fn((slug) => slug.toLowerCase().replace(' ', '-')),
}));

import { create } from 'react-test-renderer';
import React from 'react';
import 'react-native';
import { NavigationBarProductShareButton, mapStateToProps } from '../NavigationBarProductShareButton';

const props = {
	manufacturer: 'mfg',
	title: 'title',
	productId: 'productId',
	compositeId: 1234,
	type: 'type',
	selectedFinish: {},
	uniqueId: 5678,
	useBranchLinksForShare: true,
	description: 'description',
};

describe('NavigationBarProductShareButton', () => {
	it('should render with full props', () => {
		const result = create(<NavigationBarProductShareButton {...props} />).toJSON();
		expect(result).toMatchSnapshot();
	});
	describe('onSharePress', () => {

		beforeEach(() => {
			Share.open.mockClear();
		});

		it('should handle a press with uniqueId', async () => {
			const wrapper = create(<NavigationBarProductShareButton {...props} />);
			const instance = wrapper.getInstance();
			await instance.onSharePress();
			expect(branch.createBranchUniversalObject).toBeCalledWith('mfg productId title', {
				contentIndexingMode: 'private',
				title: 'mfg productId title on Build.com',
			});
			expect(Share.open).toBeCalledWith({
				message: 'View the mfg productId title on Build.com',
				title: 'Share Product',
				url: '/url',
			});
		});
		it('should handle a press without a uniqueId', async () => {
			const newProps = { ...props };
			delete newProps.uniqueId;
			newProps.useBranchLinksForShare = false;
			const wrapper = create(<NavigationBarProductShareButton {...newProps} />);
			const instance = wrapper.getInstance();
			await instance.onSharePress();
			expect(branch.createBranchUniversalObject).toBeCalledWith('mfg productId title', {
				contentIndexingMode: 'private',
				title: 'mfg productId title on Build.com',
			});
			expect(Share.open).toBeCalledWith({
				message: 'View the mfg productId title on Build.com',
				title: 'Share Product',
				url: 'https://www.build.com/mfg-productid/s1234',
			});
		});
	});
	describe('mapStateToProps', () => {

		const productsReducer = {
			1234: {
				description: 'description',
				manufacturer: 'mfg',
				productId: 'productId',
				selectedFinish: {
					uniqueId: 5678,
				},
				title: 'title',
				type: 'type',
			},
		};
		const featuresReducer = {
			features: {
				useBranchLinksForShare: true,
			},
		};

		it('should handle with product', () => {
			const result = mapStateToProps({
				productsReducer,
				featuresReducer,
			}, {
				compositeId: 1234,
			});
			expect(result).toEqual({
				description: 'description',
				manufacturer: 'mfg',
				productId: 'productId',
				selectedFinish: {
					uniqueId: 5678,
				},
				uniqueId: 5678,
				title: 'title',
				type: 'type',
				useBranchLinksForShare: true,
			});
		});
		it('should handle without selectedFinish', () => {
			delete productsReducer['1234'].selectedFinish;
			const result = mapStateToProps({
				productsReducer,
				featuresReducer: {
					features: {
						useBranchLinksForShare: true,
					},
				},
			}, {
				compositeId: 1234,
			});
			expect(result).toEqual({
				description: 'description',
				manufacturer: 'mfg',
				productId: 'productId',
				selectedFinish: undefined,
				uniqueId: undefined,
				title: 'title',
				type: 'type',
				useBranchLinksForShare: true,
			});		});
		it('should handle without product', () => {
			const result = mapStateToProps({
				productsReducer: {},
			}, {
				compositeId: 1234,
			});
			expect(result).toEqual({
				description: '',
			});
		});
	});
});
