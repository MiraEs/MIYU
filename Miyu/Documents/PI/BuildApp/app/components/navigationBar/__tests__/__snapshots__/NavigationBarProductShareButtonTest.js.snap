// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarProductShareButton should render with full props 1`] = `
<NavigationBarIconButton
  iconName="share"
  loading={false}
  onPress={[Function]}
  trackAction="PDP_NAV_TAP_SHARE"
/>
`;
